src/components/events-table.js

```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
 }
 
 const EventsTable = ({ taskData }) => {
-    let processEvents = taskData === null ? [] : taskData.processEvents;
+    const [anchorLevelEl, setAnchorLevelEl] = React.useState(null);
+    const [anchorLogEl, setAnchorLogEl] = React.useState(null);
+
+    const [processEvents] = React.useState(taskData.processEvents);
+    const [filtredProcessEvents, setFiltredProcessEvents] = React.useState(
+        taskData.processEvents
+    );
+
+    const [levelFilter, setLevelFilter] = React.useState('');
+    const [logFilter, setLogFilter] = React.useState('');
+    const handleLevelMenuClick = (event) => {
+        setAnchorLevelEl(event.currentTarget);
+    };
+
+    const handleLevelChange = (event) => {
+        setLevelFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(event.currentTarget.value.toUpperCase(), logFilter);
+    };
+
+    const handleLogMenuClick = (event) => {
+        setAnchorLogEl(event.currentTarget);
+    };
+    const handleLogChange = (event) => {
+        setLogFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(
```

Comment:
I would rather call the filter method in the rendering directly, so we don't have to define an additional state filteredProcessEvents, and we're sure we will not forget to add the filtering in the future.